npm/ipsm/ipsm.go

```diff
@@ -137,71 +196,137 @@ func (ipsMgr *IpsetManager) CreateList(listName string) error {
 		spec:          []string{util.IpsetSetListFlag},
 	}
 	log.Logf("Creating List: %+v", entry)
-	if errCode, err := ipsMgr.Run(entry); err != nil && errCode != 1 {
+	if errCode, err := ipsMgr.run(entry); err != nil && errCode != 1 {
 		metrics.SendErrorLogAndMetric(util.IpsmID, "Error: failed to create ipset list %s.", listName)
 		return err
 	}
 
-	ipsMgr.ListMap[listName] = NewIpset(listName)
+	ipsMgr.listMap[listName] = newIpset(listName)
+	return nil
+}
+
+// Destroy completely cleans ipset.
+func (ipsMgr *IpsetManager) destroy() error {
+	entry := &ipsEntry{
+		operationFlag: util.IpsetFlushFlag,
+	}
+	if _, err := ipsMgr.run(entry); err != nil {
+		metrics.SendErrorLogAndMetric(util.IpsmID, "Error: failed to flush ipset")
+		return err
+	}
+
+	entry.operationFlag = util.IpsetDestroyFlag
+	if _, err := ipsMgr.run(entry); err != nil {
+		metrics.SendErrorLogAndMetric(util.IpsmID, "Error: failed to destroy ipset")
+		return err
+	}
 
+	//TODO set IPSetInventory to 0 for all set names
 	return nil
 }
 
-// DeleteList removes an ipset list.
-func (ipsMgr *IpsetManager) DeleteList(listName string) error {
+// createSet creates an ipset.
+func (ipsMgr *IpsetManager) createSet(setName string, spec []string) error {
+	timer := metrics.StartNewTimer()
+
+	if _, exists := ipsMgr.setMap[setName]; exists {
```

Comment:
sam with this check, i think we can safely move this to the public CreateSet function